src/sentry/migrations/0514_slugify_invalid_monitors.py

```diff
@@ -0,0 +1,79 @@
+# Generated by Django 2.2.28 on 2023-07-07 15:51
+from django.db import IntegrityError, migrations
+from django.utils.text import slugify
+
+from sentry.constants import ObjectStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def migrate_monitor_slugs(apps, schema_editor):
+    Monitor = apps.get_model("sentry", "Monitor")
+    Rule = apps.get_model("sentry", "Rule")
+
+    MAX_SLUG_LENGTH = 50
+    monitors_to_clean_up = []
+
+    for monitor in RangeQuerySetWrapperWithProgressBar(Monitor.objects.all()):
+        monitor_slug = monitor.slug
+        slugified = slugify(monitor_slug)[:MAX_SLUG_LENGTH].strip("-")
+
+        # Nothing to migrate if the monitor already has a valid slug
+        if monitor_slug == slugified:
+            continue
+
+        try:
+            monitor.slug = slugified
+            monitor.save()
+        except IntegrityError:
+            # If there is a collision, delete the old monitor as the new one is receiving all check-ins
+            monitors_to_clean_up.append(monitor.id)
+
+    for delete_monitor in Monitor.objects.filter(id__in=monitors_to_clean_up):
+        alert_rule_id = delete_monitor.config.get("alert_rule_id")
+        if alert_rule_id:
+            rule = (
+                Rule.objects.filter(
+                    project_id=delete_monitor.project_id,
+                    id=alert_rule_id,
+                )
+                .exclude(
+                    status__in=[
+                        ObjectStatus.PENDING_DELETION,
+                        ObjectStatus.DELETION_IN_PROGRESS,
+                    ]
+                )
+                .first()
+            )
+            if rule:
+                rule.status = ObjectStatus.PENDING_DELETION
+                rule.save()
+
+        delete_monitor.status = ObjectStatus.PENDING_DELETION
```

Comment:
will this eventually be cleaned up? I can't seem to call `ScheduledDeletion.schedule()` directly in here as it's a class method

Reply:
If you're using scheduled deletions you'll need to create the rows. @ceorourke  did this recently for `Rule` deletions https://github.com/getsentry/sentry/blob/8d4b9ccb38443ebd9d209c7f40b38f1917a191be/src/sentry/migrations/0507_delete_pending_deletion_rules.py#L19-L43

Reply:
Regionless is fine for now.